pkg/controller/node/nodecontroller.go

```diff
@@ -565,6 +677,10 @@ func (nc *NodeController) monitorNodeStatus() error {
 			continue
 		}
 		node := nodeCopy.(*v1.Node)
+
+		// taint or clear NoSchedule taint based on condition change, we should do this before tryUpdateNodeStatus
+		nc.addOrRemoveNoScheduleTaintByCondition(node)
```

Comment:
The only way to eliminate this return value is verifying contents in `noScheduleTainer`, however, this is still tricky because you actually need to verify contents change in `noScheduleTainer`.